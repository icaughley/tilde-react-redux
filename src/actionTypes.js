export const ERROR_SUFFIX = '_ERROR';

export const GET_PROJECTS = 'GET_PROJECTS';
export const CREATE_PROJECT = 'CREATE_PROJECT';
export const UPDATE_PROJECT = 'UPDATE_PROJECT';
export const DELETE_PROJECT = 'DELETE_PROJECT';
export const SET_PROJECT_CLOAKED = 'SET_PROJECT_CLOAKED';
export const GET_INVOICING = 'GET_INVOICING';
export const GET_BILLABLE_PROJECTS = 'GET_INVOICING_PROJECTS';
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';
export const OPEN_MESSAGE = 'OPEN_MESSAGE';
export const CLOSE_MESSAGE = 'CLOSE_MESSAGE';
export const SET_SELECTED_PROJECT = 'SET_SELECTED_PROJECT';
export const GET_WORK = 'GET_WORK';
