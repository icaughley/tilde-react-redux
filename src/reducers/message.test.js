import deepFreeze from "deep-freeze";
import reduce from "./message";
import {CLOSE_MESSAGE, OPEN_MESSAGE} from "../actionTypes";

describe('Message reducer', () => {
    it('provides the initial state', () => {
        expect(reduce()).toEqual({});
    });

    it('ignores unrecognised actions', () => {
        expect(reduce({}, {type: 'UNRECOGNISED'})).toEqual({});
    });

    it('handles action of type OPEN_MESSAGE with success message', () => {
        const action = {type: OPEN_MESSAGE, payload: {type: "success", msg: "The message"}};
        const state = {};

        deepFreeze(state);
        deepFreeze(action);

        expect(reduce(state, action))
            .toEqual({
                open: true,
                icon: "check circle",
                header: "Success",
                type: "success",
                msg: "The message"
            });
    });

    it('handles action of type OPEN_MESSAGE with error message', () => {
        const action = {type: OPEN_MESSAGE, payload: {type: "error", msg: "The message"}};
        const state = {};

        deepFreeze(state);
        deepFreeze(action);

        expect(reduce(state, action))
            .toEqual({
                open: true,
                icon: "warning sign",
                header: "Error",
                type: "error",
                msg: "The message"
            });
    });

    it('handles action of type CLOSE_MESSAGE', () => {
        const action = {type: CLOSE_MESSAGE, payload: {type: "success", msg: "The message"}};
        const state = {
            open: true,
            icon: "warning sign",
            header: "Error",
            type: "error",
            msg: "The message"
        };

        deepFreeze(state);
        deepFreeze(action);

        expect(reduce(state, action))
            .toEqual({
                open: false,
                icon: "warning sign",
                header: "Error",
                type: "error",
                msg: "The message"
            });
    });
});
