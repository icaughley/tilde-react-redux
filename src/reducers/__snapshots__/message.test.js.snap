exports[`Message reducer handles action of type CLOSE_MESSAGE 1`] = `
Object {
  "header": "Error",
  "icon": "warning sign",
  "msg": "The message",
  "open": false,
  "type": "error",
}
`;

exports[`Message reducer handles action of type OPEN_MESSAGE with error message 1`] = `
Object {
  "header": "Error",
  "icon": "warning sign",
  "msg": "The message",
  "open": true,
  "type": "error",
}
`;

exports[`Message reducer handles action of type OPEN_MESSAGE with success message 1`] = `
Object {
  "header": "Success",
  "icon": "check circle",
  "msg": "The message",
  "open": true,
  "type": "success",
}
`;
